### PR TITLE
Add `public` qualifier for some member functions in Java class `Schema`

### PR DESCRIPTION
**Pull request by ttnghia**
This adds the public qualifier for some member functions of `Schema` class in Java code, allowing them to be accessed outside of the `ai.rapids.cudf` package such as from spark-rapids-jni or Spark plugin.

Java docs are also added for the newly became public functions as well as some existing public functions.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Schema.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ private void flattenIfNeeded() {
 
   private int flattenedLength(int startingLength) {
     if (childSchemas != null) {
-      for (Schema child: childSchemas) {
+      for (Schema child : childSchemas) {
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/Schema.java:123]: The IDE formats the entire file, which causes some unrelated changes.

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Schema.java:123]: Can you please remove the changes or update the IDE then?

File: java/src/main/java/ai/rapids/cudf/Schema.java [hunk 2]
```diff
@@ -182,7 +182,7 @@ public boolean hasNestedChildren() {
     return false;
   }
 
-  int[] getFlattenedTypeIds() {
+  public int[] getFlattenedTypeIds() {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Schema.java:185]: Can you add some javadocs to these methods that are now public. If they are public at a minimum we have to document how they are flattened so that a user could make use of it.

File: java/src/main/java/ai/rapids/cudf/Schema.java (hunk 1 above)
Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/Schema.java:123]: I prefer to keep the changes as they just reformat the (incorrect) code style which was not formatted properly.

File: java/src/main/java/ai/rapids/cudf/Schema.java (hunk 2 above)
Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/Schema.java:201]: I've added Java docs for the changed functions. However, there are other similar public functions don't have Java docs thus I've also added docs for them for consistency.

File: java/src/main/java/ai/rapids/cudf/Schema.java [hunk 3]
```diff
@@ -194,6 +210,10 @@ public int[] getFlattenedTypeIds() {
     return ret;
   }
 
+  /**
+   * Get scales of the columns' types flattened from all levels in schema.
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/Schema.java:214]: nit: the key words I think we are missing are depth first.  Otherwise it is not clear how it was flattened.

**Review by revans2 — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/Schema.java (hunk 3 above)
Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/Schema.java]: Updated, adding "depth-first traversal".

File: java/src/main/java/ai/rapids/cudf/Schema.java (hunk 1 above)
Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Schema.java:123]: It was formatted properly. The space in the for loop is not part of the formatting guidelines. It is something you IDE decided it should add.
